### PR TITLE
Nuclear Operatives Holosuits and Strategic Station Holomap

### PR DESCRIPTION
**Pull request by JMWTurner**
Reminder that the holosuits are destroyed if their wearer's explosive implant goes off.

holomap

holosuits

:cl:
- rscadd: Added the Strategic Station Holomap. Unlike the regular Station Holomaps, this one features markers that reveal the locations of the Captain's Office, the AI chamber, the Armory, and the SMES.
- rscadd: The Tactical Turtlenecks in the lockers on the Syndicate Ship have fitted with Holosuit Chips that on top of showing each others' locations, show the location of the Ship itself (if it's on their ZLevel)

### CODE REVIEW

**Comment by JMWTurner** [comment]
revision 2 to 5
- Added Tcomms Chamber to the Strategic Station Holomap markers

**Comment by clusterfack** [comment]
Yeah I don't think nuke ops should get this

**Comment by jakmak3** [comment]
cluster what do you mean. Being able to locate your fellow operatives without having to manually call them out on the radio is a amazing ability that would open the gateway to nuke op teams cooperating and working together more efficiently. Even having to have to spend TC on these means that more likley than not 2-3 ops will even bother getting it at the most and would serve less of a purpose in general. If your worried about the crew getting these the explosive implant that nukies gets ensures that this wouldn't fall into the hands of the clown Besides this one of the main attractions to the entire holomap thing.

**Comment by clusterfack** [comment]
Well obviously you'd buy them as a pack not individually. It'd be silly if you bought them one at a time.

**Comment by JMWTurner** [comment]
> it was initially just meant for deathsquads

Actually no. Deathsquads were the first to use them so I could test them and fix bugs without affecting rounds too much. Back then people already wanted me to make holosuits for nuke ops, ert, and vox raiders, and the holosuit chips were the missing link I needed to figure out to allow it.

I have no plans for anyone in the station crew having them though. As @jakmak3 said I think they suit small operative teams just fine.

Also https://github.com/d3athrow/vgstation13/blob/Bleeding-Edge/code/modules/html_interface/map/station_map.dm#L7

**Comment by Exxion** [comment]
How does this even really affect balance?
It does two things:
1. Gives the ops a map, which they could already get by clicking a button in the top right
2. Lets them see where their comrades and ship are, which any good op already knows

This just makes the gamemode easier on new players, which is always a plus

File: code/modules/power/smes.dm [hunk 1]
```diff
@@ -1,6 +1,8 @@
 // the SMES
 // stores power
 
+var/list/smess = list()
```
Comment (clusterfack) [inline code/modules/power/smes.dm]: Don't use crappy short names like this for global variables

File: code/game/machinery/station_map.dm [hunk 2]
```diff
@@ -329,3 +329,93 @@ var/list/station_holomaps = list()
...
+	density = 1
+	machine_flags = null
+	icon = 'icons/obj/stationobjs_64x64.dmi'
+	pixel_x = -1*WORLD_ICON_SIZE/2
+	pixel_y = -1*WORLD_ICON_SIZE/2
+
+	var/list/watching_mobs = list()
+	var/list/watcher_maps = list()
+
+/obj/machinery/station_map/strategic/New()
```
Comment (clusterfack) [inline code/game/machinery/station_map.dm]: Fairly annoyed that I have to ask this EVERY time but where is the destroy proc

Comment (JMWTurner) [reply code/game/machinery/station_map.dm:346]: I don't remember it being a rule that every atom needs a custom Destroy() proc but I added one bellow if it pleases you.

Comment (clusterfack) [inline code/game/machinery/station_map.dm]: The rule has ALWAYS been if it holds references it needs to have those references removed in destroy

File: code/game/machinery/station_map.dm [hunk 3]
```diff
@@ -348,6 +348,9 @@ var/list/station_holomaps = list()
 	if(ticker && holomaps_initialized)
 		initialize()
 
+/obj/machinery/station_map/strategic/Destroy()
+	..()
```
Comment (Exxion) [inline code/game/machinery/station_map.dm]: Wow, it's fucking nothing

Comment (Exxion) [reply code/game/machinery/station_map.dm]: OK I guess just memeing isn't very helpful so let me clarify:
He didn't mean add it just for the sake of a nonexistent standard. It has to remove any references the object holds and any references other things hold to it.
The parent should do that for... well, the parent object. You just need to add code here to null out anything specific to this object. Such as that datum created in `New()`.

File: code/game/machinery/station_map.dm [hunk 4]
```diff
@@ -329,3 +329,96 @@ var/list/station_holomaps = list()
...
+
+/obj/machinery/station_map/strategic/New()
+	..()
+	holomap_datum = new /datum/station_holomap/strategic()
+	original_zLevel = map.zMainStation
+	if(ticker && holomaps_initialized)
+		initialize()
+
+/obj/machinery/station_map/strategic/Destroy()
+	..()
```
Comment (Intigracy) [inline code/game/machinery/station_map.dm]: As amusing as you being a smartass with this is, set holomap_datum = null

Comment (Exxion) [reply code/game/machinery/station_map.dm]: Gee Bill, how come your mom lets you have TWO comments on a line?

Comment (Intigracy) [reply code/game/machinery/station_map.dm]: You borked it buddy, yours doesn't show on the files changed tab.

Comment (Exxion) [reply code/game/machinery/station_map.dm]: God damn it, GitHub
Oh well, guess emphasis can't hurt

File: code/game/machinery/station_map.dm (hunk 2 above)
Comment (JMWTurner) [reply code/game/machinery/station_map.dm:346]: oh, then it was already missing in the parent Destroy(). I added it there.

Comment (JMWTurner) [reply code/game/machinery/station_map.dm:346]: stopWatching() takes care of that, and is called by the parent Destroy()

**Comment by JMWTurner** [comment]
Is it good now?

**Comment by ComicIronic** [comment]
As a rule, you should split the content and mapping parts of any PR into two, because mapping changes are a lot harder to get through if there's code review, and code changes are harder to get through if you are constantly getting map conflicts.

**Comment by JMWTurner** [comment]
The only changes done were to the syndicate bases, I doubt it'd cause conflict for anyone right now.

**Comment by JMWTurner** [comment]
@ComicIronic There you go #12539 I removed the map changes from this PR.

**Comment by Megamaw** [comment]
instead of making them the ones in the lockers, can you just make it so its the ones you start out with b/c it'll be hard to tell which turtleneck has the badge at a glance, when everyone's changing clothes and putting their initial stuff away

**Comment by JMWTurner** [comment]
@Megamaw I have to disagree.

chip

They have different names too (Tactical Turtleneck, Tactical Holosuit). Nuke ops are going to pick them up when they dress up anyway. At least they're more likely to remember to pick those up than pinpointers.

**Comment by Megamaw** [comment]
ah, got it! good to know.

**Comment by JMWTurner** [comment]
Bump

**Comment by Intigracy** [comment]
Changelog is out of date from the look of it.

**Comment by JMWTurner** [comment]
I removed the part of the changelog that said it was already on the maps. That's what #12539 will be for.

**Comment by JMWTurner** [comment]
B-Bump?

**Comment by jakmak3** [comment]
yeah whens this shit getting merged

**Comment by Intigracy** [comment]
rn

**Pull request merged**